bundles/automation/org.eclipse.smarthome.automation.json.parsers/src/main/java/org/eclipse/smarthome/automation/parser/json/ConditionTypeJSONParser.java

```diff
@@ -0,0 +1,181 @@
+
+/**
+ * @author Ana Dimova - Initial Contribution
+ *
+ */
+public class ConditionTypeJSONParser {
+
+    /**
+     * This method is used for reversion of {@link ConditionType} to JSON format.
+     *
+     * @param conditionType is a {@link ConditionType} object to revert.
+     * @return JSONObject is an object representing the {@link ConditionType} in json format.
+     * @throws IOException
+     * @throws JSONException
+     */
+    static void conditionTypeToJSON(ConditionType conditionType, OutputStreamWriter writer) throws IOException,
+            JSONException {
+        ModuleTypeJSONParser.moduleTypeToJSON(conditionType, writer);
+        Set<Input> inputs = conditionType.getInputs();
+        if (inputs != null && !inputs.isEmpty()) {
+            Set<ConfigDescriptionParameter> configDescriptions = conditionType.getConfigurationDescription();
+            if (configDescriptions != null && !configDescriptions.isEmpty())
+                writer.write(",\n    \"" + JSONStructureConstants.INPUT + "\":{\n");
+            Iterator<Input> inputsI = inputs.iterator();
+            while (inputsI.hasNext()) {
+                Input input = inputsI.next();
+                writer.write("      \"" + input.getName() + "\":{\n");
+                InputJSONParser.inputToJSON(input, writer);
+                if (inputsI.hasNext())
+                    writer.write("\n      },");
+                else
+                    writer.write("\n      }");
+            }
+            writer.write("\n    }");
+        }
+    }
+
+    /**
+     * This method is used for reversion of {@link CompositeConditionType} to JSON format.
+     *
+     * @param cConditionType is a {@link CompositeConditionType} object to revert.
+     * @return JSONObject is an object representing the {@link CompositeConditionType} in json format.
+     * @throws IOException
+     * @throws JSONException
+     */
+    static void compositeConditionTypeToJSON(CompositeConditionType cConditionType, OutputStreamWriter writer,
+            ModuleTypeManager moduleManager) throws IOException, JSONException {
+        conditionTypeToJSON(cConditionType, writer);
+        List<Condition> conditions = cConditionType.getModules();
+        Set<ConfigDescriptionParameter> configDescriptions = cConditionType.getConfigurationDescription();
+        if (configDescriptions != null && !configDescriptions.isEmpty())
+            writer.write(",\n");
+        writer.write("    " + JSONStructureConstants.CONDITIONS + ":{\n");
+        ModuleJSONParser.writeModules(moduleManager, conditions, writer);
+        writer.write("    }");
+    }
+
+    /**
+     *
+     * @param connections
+     * @param condition
+     * @param triggersMap
+     * @param moduleManager
+     * @throws ClassNotFoundException
+     * @throws IllegalArgumentException
+     */
+    static boolean validateConditionConnections(BundleContext bc, Set<Connection> connections, Condition condition,
```

Comment:
This is a validation of the content, which I think should not be within the parser, but rather in the provider. The parser should imho concentrate on the syntax.
Moving the validation logic also makes a lot of sense because it would automatically e in place for any other parser as well and the logic must not be duplicated.